shared/ui-components/src/forms/Field.tsx

```diff
@@ -24,37 +26,56 @@ const Field = (props: FieldProps) => {
   if (props.error) {
     classes.push("error")
   }
+
   const labelClasses = ["label"]
-  if (props.caps) {
-    labelClasses.push("field-label--caps")
-  }
+  if (props.caps) labelClasses.push("field-label--caps")
+  if (props.primary) labelClasses.push("text-primary")
+  if (props.readerOnly) labelClasses.push("sr-only")
+
   const controlClasses = ["control"]
   if (props.controlClassName) {
     controlClasses.push(props.controlClassName)
   }
 
+  const type = props.type || "text"
+  const isRadioOrCheckbox = ["radio", "checkbox"].includes(type)
+
+  const label = useMemo(
+    () => (
+      <label className={labelClasses.join(" ")} htmlFor={props.id || props.name}>
+        {props.label}
+      </label>
+    ),
+    [props.label, props.id, props.name, labelClasses]
+  )
```

Comment:
Why is this memoized? I don't _think_ it'll cause problems, but it just seems to add unnecessary complication to me, it's not a particularly expensive operation...

Reply:
hmm, in general, it's not an expensive operation, but for each situation when e.g user submit a form and there is an error state - each label is rendered again. useMemo prevents unnecessary re-render.